src/generator/generator.ts

```diff
@@ -69,6 +68,33 @@ function isSimpleType(type: string): boolean {
   return true;
 }
 
+/**
+ * Attempt to turn a regex into a more human readable form.
+ * @param regex pattern for the given parameter
+ */
+function unRegex(regex: string): string {
+  // example: ^projects/[^/]+$' ==> projects/my-project
+  let pattern = regex;
+  if (typeof regex !== 'string') {
+    return '';
+  }
+  // remove leading ^
+  if (pattern.startsWith('^')) {
+    pattern = pattern.slice(1);
+  }
+  // remove trailing $
+  if (pattern.endsWith('$')) {
+    pattern = pattern.slice(0, pattern.length - 1);
+  }
+  // replace projects placeholders
+  pattern = pattern.replace('projects/[^/]+', 'projects/my-project');
+  pattern = pattern.replace('locations/[^/]+', 'locations/my-location');
+  // TODO: This should be replaced with a proper regular express
+  // that does a look behind to find the name to be replaced.  Alex
+  // will probably know how to do it.
```

Comment:
Oh, you don't need look-behind here. You only need look-behinds when the string being looked-behind is unknown. Here, it's easier:

```
> pattern = "^projects/[^/]+/locations/[^/]+/products/[^/]+/referenceImages/[^/]+$"
'^projects/[^/]+/locations/[^/]+/products/[^/]+/referenceImages/[^/]+$'
> pattern.replace(/\^?(\w+)s\/\[\^\/\]\+\$?/g, "$1s/my-$1")
'projects/my-project/locations/my-location/products/my-product/referenceImages/my-referenceImage'
```

You only need to make sure that different regex applications do not overlap, other than that, it's pretty straightforward, with some added backslashes. We just hope they have all plurals ending with "s", I can't singularize in regex properly :)

```
description:   optional ^     project  s   /  [   ^   /   ]  +  optional $
      regex: /     \^?        (\w+)    s  \/ \[  \^  \/  \] \+     \$?      /
```
and then we replace it with `$1s/my-$1` which expands to `projects/my-project`.

Reply:
This worked like a dream.  Thank you!